lib/spack/spack/schema/modules.py

```diff
@@ -257,4 +320,7 @@ def update(data):
         changed = True
         data['default'] = default
 
+    # translate exclude/include to blacklist/whitelist
+    changed |= update_keys(data, exclude_include_translations)
```

Comment:
If we hit the early exit at line 310, we'll never reach this update.

Reply:
fixed

Reply:
Making this work revealed that the `module_configuration` fixture needs to use `mutable_config`.